Source/MethodCall.cs

```diff
@@ -340,20 +309,20 @@ public ICallbackResult Callback(Delegate callback)
 
 		protected virtual void SetCallbackWithoutArguments(Action callback)
 		{
-			this.setupCallback = delegate { callback(); };
+			this.callbackResponse = (object[] args) => callback();
```

Comment:
When I first read that code, I was wondering why old-style `delegate` was used here. Make the lambda parameter explicit so it's clear exactly what is going on here, and also to make this method look more similar to `SetCallbackWithArguments` (below).